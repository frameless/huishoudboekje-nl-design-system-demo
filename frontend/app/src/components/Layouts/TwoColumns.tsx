import React from "react";
import {useIsMobile} from "react-grapple";
import {TABLET_BREAKPOINT} from "../../utils/things";
import {Flex, Stack} from "@chakra-ui/react";
import Logo from "../Logo";

const TwoColumns = ({children}) => {
	const isMobile = useIsMobile(TABLET_BREAKPOINT);

	return (
		<Flex h={"auto"} minHeight={"100vh"} minWidth={"100%"} w={"auto"} bg={isMobile ? "white" : "gray.100"}>
			<Stack width={"100%"} direction={isMobile ? "column" : "row"} justifyContent={"center"} alignItems={"center"} spacing={30}>
				<Stack bg={"grey.800"} height={isMobile ? "auto" : "100%"} alignItems={isMobile ? "center" : "flex-end"} justifyContent={"center"}
				       width={isMobile ? "100%" : "50%"}
				       p={5}>
					<Logo as={Flex} maxWidth={400} width={"100%"} />
				</Stack>
				<Stack bg={"white"} height={isMobile ? "auto" : "100%"} justifyContent={"center"} width={isMobile ? "100%" : "50%"} p={5}>
					{children}
				</Stack>
			</Stack>
		</Flex>
	);
};

export default TwoColumns;