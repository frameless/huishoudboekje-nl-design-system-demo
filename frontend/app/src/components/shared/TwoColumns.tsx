import {Flex, Stack} from "@chakra-ui/react";
import React from "react";
import Logo from "./Logo";
import VersionTag from "./VersionTag";

const TwoColumns = ({children}) => {
	return (
		<Flex h={"auto"} minHeight={"100vh"} minWidth={"100%"} w={"auto"} bg={["white", null, null, "gray.100"]}>
			<Stack width={"100%"} direction={["column", null, null, "row"]} justifyContent={"center"} alignItems={"center"} spacing={30}>
				<Stack bg={"grey.800"} height={["auto", null, null, "100%"]} alignItems={["center", null, null, "flex-end"]} justifyContent={"center"}
					width={["100%", null, null, "50%"]}
					p={5}>
					<Flex maxWidth={400} width={"100%"} height={"auto"} maxHeight={500}>
						<Logo />
					</Flex>
				</Stack>
				<Stack bg={"white"} height={["auto", null, null, "100%"]} justifyContent={"center"} width={["100%", null, null, "50%"]} p={5}>
					<Stack flex={1} justify={"center"}>
						<Stack>
							{children}
						</Stack>
					</Stack>
					<Stack align={["center", null, null, "flex-start"]} pt={[10, null, null, 0]}>
						<VersionTag />
					</Stack>
				</Stack>
			</Stack>
		</Flex>
	);
};

export default TwoColumns;