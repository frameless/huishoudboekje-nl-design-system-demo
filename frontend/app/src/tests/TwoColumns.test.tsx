import {Text} from "@chakra-ui/react";
import {getByText, render} from "@testing-library/react";
import React from "react";
import TwoColumns from "../components/shared/TwoColumns";

jest.mock("react-i18next", () => require("./utils/mock-hooks").reactI18NextMock());
jest.mock("react-router-dom", () => require("./utils/mock-hooks").reactRouterDomMock());

describe("TwoColumns", () => {

	it("Renders the children", async () => {
		const {container} = render((
			<TwoColumns>
				<Text>Children</Text>
			</TwoColumns>
		));

		expect(container.innerHTML).toMatchSnapshot();

		const text = getByText(container, "Children");
		expect(text).toBeVisible();
	});

});