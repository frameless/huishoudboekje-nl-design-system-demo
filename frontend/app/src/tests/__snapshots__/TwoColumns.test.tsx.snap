// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TwoColumns Renders the children 1`] = `"<div class=\\"css-ivkdgu\\"><div class=\\"chakra-stack css-1czofa7\\"><div class=\\"chakra-stack css-13z5mb4\\"><div class=\\"css-yjnu3x\\"><div to=\\"/\\" class=\\"css-4dhf1d\\"><img alt=\\"logo\\" src=\\"/theme/logo.svg\\" class=\\"chakra-image css-0\\"></div></div></div><div class=\\"chakra-stack css-k49deb\\"><div class=\\"chakra-stack css-16hczwi\\"><div class=\\"chakra-stack css-n21gh5\\"><p class=\\"chakra-text css-0\\">Children</p></div></div><div class=\\"chakra-stack css-nj1o9c\\"><div class=\\"chakra-stack css-1qpox0b\\"><p class=\\"chakra-text css-qljqz\\">version</p></div></div></div></div></div>"`;
